### ISSUE TITLE
spacing around fractions is incorrect

### ISSUE THREAD

**Issue by kevinbarabash** [Khan/KaTeX#249]
The image diff below shows KaTeX rendering in black and LaTeX rendering in white.  Notice that spacing around the fractions and solidus don't match up.
![b6ed6322-14fd-11e5-843f-cc2814e0d921](https://cloud.githubusercontent.com/assets/1044413/8215794/bf07b372-14ef-11e5-8bc4-0b54c788694f.png)

I'll post a minimal repro case later.

**Comment by kevinb7** [comment]
Credit goes to @gagern for finding this issue.  I really like idea of these images diffs to compare KaTeX output to LaTeX.

**Comment by gagern** [comment]
Using the image diff format from #268, the diff now looks like this (green is LaTeX, red is KaTeX):

![arrays-diff](https://cloud.githubusercontent.com/assets/190759/8351742/d7bec252-1b2f-11e5-8f3d-5648b90960c8.png)

The difference in the `NestedFractions` test is probably due to the same problem:

![nestedfractions-diff](https://cloud.githubusercontent.com/assets/190759/8351764/f92d9472-1b2f-11e5-9382-24381a409e25.png)

**Comment by kevinb7** [comment]
I noticed that the our fraction bars aren't wide enough in some cases when I was working on ```\cfrac```, see https://github.com/Khan/KaTeX/pull/135.

**Comment by gagern** [comment]
OK, I did some research on this. TeX puts 1.2pt of space before and after every fraction. This is the value of `\nulldelimiterspace`. So the branches of the `genfrac` handler where it adds delimiters should add such null delimiters if no actual delimiters are given. The `leftright` handler already adds null delimiters, so we can duplicate functionality from there.

I also noticed that when I apply `\showlists` to some fraction, each fraction is contained in a `\mathord`, and is not simply of type Inner. This is probably due to the fact that the definition of `\frac` reads

```tex
\DeclareRobustCommand{\frac}[2]{{\begingroup#1\endgroup\@@over#2}}
```

which has two sets of braces, so it forms a brace-delimited group which gets translated into a `\mathord`.

Both the `\nulldelimiterspace` and the mechanism turning the Inner into an Ord are discussed in https://github.com/mathjax/MathJax/issues/969, with valuable explanations from David Carlisle.
